server.js

```diff
@@ -47,6 +47,228 @@ app.get('/authenticate', (request, response) => {
+      }
+      return response.status(404).json({
+        error: 'We could not find that student!'
+      })
+    })
+    .catch(error => {
+      response.status(500).json({ error })
+    });
+});
+
+app.get('/api/v1/programs/:id', (request, response) => {
+  const { id } = request.params;
+
+  database('programs').select()
+    .then(program => {
+      if (program) {
+        response.status(200).json(program)
+      }
+      return response.status(404).json({
+        error: 'There were no program found!'
+      })
+    })
+    .catch(error => {
+      response.status(500).json({ error })
+    });
+});
+
+app.post('/api/v1/mentors', (request, response) => {
+
+  const mentor = request.body;
+
+  database('mentors').insert(mentor, 'id')
+    .then(mentor => {
+      if (result) {
+        resonse.status(201).json({ id: mentor[0] });
+      } else {
+        response.status(201).json({ error: 'You are missing a property'});
+      }
+    })
+    .catch(error => {
+      response.status(500).json({ error });
+    });
+});
+
+app.post('/api/v1/students', (request, response) => {
+
+  const student = request.body;
+
+  database('students').insert(student, 'id')
+    .then(student => {
+      if (result) {
+        resonse.status(201).json({ id: student[0] });
+      } else {
+        response.status(201).json({ error: 'You are missing a property'});
+      }
+    })
+    .catch(error => {
+      response.status(500).json({ error });
+    });
+});
+
+app.post('/api/v1/programs', (request, response) => {
+
+  const program = request.body;
+
+  database('programs').insert(program, 'id')
+    .then(program => {
+      if (result) {
+        resonse.status(201).json({ id: program[0] });
+      } else {
+        response.status(201).json({ error: 'You are missing a property'});
+      }
+    })
+    .catch(error => {
+      response.status(500).json({ error });
+    });
+});
+
+app.patch('/api/v1/mentors/:id', (request, response) => {
+  const { id } = request.params;
+  const update = request.body;
+
+  database('mentors').where('id', id).update(update)
+    .then(update => {
+      response.status(200).json({ update });
+    })
+    .catch(error => {
+      response.status(500).json({ error });
+    });
+});
+
+app.patch('/api/v1/students/:id', (request, response) => {
+  const { id } = request.params;
+  const update = request.body;
+
+  database('students').where('id', id).update(update)
+    .then(update => {
+      response.status(200).json({ update });
+    })
+    .catch(error => {
+      response.status(500).json({ error });
+    });
+});
+
+app.patch('/api/v1/programs/:id', (request, response) => {
+  const { id } = request.params;
+  const update = request.body;
+
+  database('programs').where('id', id).update(update)
+    .then(update => {
+      response.status(200).json({ update });
+    })
+    .catch(error => {
+      response.status(500).json({ error });
+    });
+});
+
+app.delete('/api/v1/mentors/:id', (request, response) => {
```

Comment:
Do we want to secure this? Maybe tie that in with admin access only? So only Allison can delete mentors by request?